### PR TITLE
Split the tfvars handling into a separate syntax file

### PR DESCRIPTION
**Pull request by rchl**
The main syntax no longer handles the .tfvars extension. Handling of
.tfvars is now done by the new syntax file which also assigns a new
source.terraform-vars scope instead of the previous source.terraform.

This change is to allow for differentiating .tfvars and .tf files when
using a terraform-ls language server.

The .tmLanguage syntax files are not changed in the same way as those
don't support including other syntaxes so would have to be duplicated
instead. And since those are long obsolete (in ST at least), I didn't
see much point in doing anything with them.

Fixes #49

### CODE REVIEW

**Comment by rchl** [comment]
> The .tmLanguage syntax files are not changed in the same way as those
> don't support including other syntaxes

Not 100% sure about that... If the including is possible then I could consider handling those too.

As for tests for tfvars file, I've just duplicated the existing one. That new syntax file could be trimmed to include tests for features that are specific to tfvars files but the syntax supports everything anyway so not sure.

File: Terraform-tfvars.sublime-syntax [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# https://github.com/hashicorp/hcl2/blob/master/hcl/hclsyntax/spec.md
+#
+# For documentation on the .subline-syntax format:
+# https://www.sublimetext.com/docs/syntax.html
+#
+# Regex's in this file support the Oniguruma regex engine:
+# https://raw.githubusercontent.com/kkos/oniguruma/5.9.6/doc/RE
+#
+---
+name: Terraform (vars)
```
Comment (radeksimko) [inline Terraform-tfvars.sublime-syntax:16]:
```suggestion
name: Terraform (variables)
```

**Review by radeksimko — COMMENTED** [review]
> The .tmLanguage syntax files are not changed in the same way as those don't support including other syntaxes so would have to be duplicated instead.

That's sad but understandable, given the age of the format and its fading support in ST.

One more question:

Is there any convention for the `*.sublime-settings` and `*.sublime-syntax` files? I would expect these to follow Language ID, rather than file extensions? i.e. I'd expect `terraform-vars.sublime-settings` and `terraform-vars.sublime-syntax`, rather than tfvars.

File: Terraform-tfvars.sublime-syntax (hunk 1 above)
Comment (rchl) [reply Terraform-tfvars.sublime-syntax]: I was contemplating this and file names myself but I think that your suggestions work just fine.

Here I've opted for something shorter to take less space in the status bar but either works really.

Comment (radeksimko) [reply Terraform-tfvars.sublime-syntax]: Oh I see, I haven't considered the UI aspect of this particular wording

Comment (rchl) [reply Terraform-tfvars.sublime-syntax]: Your suggestion is fine. I don't like abbreviates either way.

**Review by radeksimko — APPROVED** [review]
Thanks for the PR.

I can confirm that it works nicely with the LSP package:

Screenshot 2021-06-21 at 11 11 24

**Comment by radeksimko** [comment]
Worth noting that I had to update the client settings within the LSP plugin settings:

```json
	"terraform":
		{
			"enabled": true,
			"languages": [
				{
					"languageId": "terraform",
					"document_selector": "source.terraform"
				},
				{
					"languageId": "typescript",
					"document_selector": "source.terraform-vars"
				}
			],
			"syntaxes":
			[
				"Packages/Terraform/Terraform.sublime-syntax"
			],
			"command": [
				"terraform-ls",
				"serve"
			],
		}
```

**Comment by rchl** [comment]
Created https://github.com/sublimelsp/LSP-terraform/issues/1 for updating LSP-terraform.

(You've accidentally used `typescript` language ID in your config)

**Comment by rchl** [comment]
BTW. In ST4 it should work to remove the `languages` and `syntaxes` settings and use `"selector": "source.terraform | source.terraform-vars"` instead.

**Pull request merged**